File: 2103-rings-and-rods/2103-rings-and-rods.py
class Solution:
    def countPoints(self, rings: str) -> int:
        ans = 0
        hash_map = {}
        for i in range(10):
            hash_map[i] = [0,0,0]
        color_map = {
            "R": 0,
            "G": 1,
            "B": 2
        }
        for i in range(0,len(rings),2):
            color = rings[i]
            index = int(rings[i+1])
            if hash_map[index][color_map[color]] == 0:
                hash_map[index][color_map[color]] = 1
        
        for key in hash_map:
            if sum(hash_map[key]) == 3:
                ans += 1
        
        return ans

File: 2103-rings-and-rods/2103-rings-and-rods.go
func countPoints(rings string) int {
    
    type Color struct {
        blue bool
        green bool
        red bool
    }
    hashMap := make(map[byte]Color)
    
    for i:= 0; i < len(rings); i += 2 {
        color := rings[i]
        num := rings[i+1]
        _, ok := hashMap[num]
        if !ok {
            hashMap[num] = Color{}
        }
        if value, ok := hashMap[num]; ok {
            switch color {
            case 'B':
                value.blue = true
            case 'R':
                value.red = true
            case 'G':
                value.green = true
            }
            hashMap[num] = value
        }
    }
    
    answer := 0
    
    for _, v := range hashMap {
        if v.blue && v.green && v.red {
            answer++
        }
    }
    
    return answer
}
